server/src-lib/Hasura/Server/Utils.hs

```diff
@@ -111,3 +113,16 @@ _2 (_, y, _) = y
 
 _3 :: (a, b, c) -> c
 _3 (_, _, z) = z
+
+-- regex related
+matchRegex :: String -> Bool -> T.Text -> Either String Bool
```

Comment:
Text -> Bool -> Text -> Either String Bool

Use the Bytestring module of TDFA instead of String internally.